network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -97,6 +67,8 @@ func DefaultMessageIDFunction(msg *pb.Message) string {
 	return h.SumHash().Hex()
 }
 
+type CreateNodeFunc func(logger zerolog.Logger, host host.Host, pubSub *pubsub.PubSub, routing routing.Routing, pCache *p2pnode.ProtocolPeerCache, uniMgr *unicast.Manager) p2pnode.LibP2PNode
+
```

Comment:
The reason we cannot set the topic validator in this fashion is because the TopicValidator is somewhat dynamic. Meaning, it depends on things that are set after node start up. For example the [authorized sender validator](https://github.com/onflow/flow-go/blob/khalil/6393-corrupted-libp2pnode/network/p2p/middleware/middleware.go#L279) is only created after the middleware starts because it depends on the m.ov.identity, we also use different [peerFilter](https://github.com/onflow/flow-go/blob/khalil/6393-corrupted-libp2pnode/network/p2p/middleware/middleware.go#L495) depending on if the channel is public or not. Because the TopicValidator is not static and can't just be created once we need to override the Subscribe method of the libP2PNode so that we can provide a no-op TopicValidator and ignore the TopicValidator passed in by the middleware during calls to Subscribe. This is why we need the boilerplate.